src/plugins/controls/public/range_slider/components/range_slider_control.tsx

```diff
@@ -6,117 +6,52 @@
  * Side Public License, v 1.
  */
 
-import React, { FC, useState, useRef } from 'react';
+import { debounce } from 'lodash';
+import React, { FC, useState, useRef, useMemo, useEffect } from 'react';
 
-import {
-  EuiFieldNumber,
-  EuiText,
-  EuiInputPopover,
-  EuiLoadingSpinner,
-  EuiFlexGroup,
-  EuiFlexItem,
-} from '@elastic/eui';
+import { EuiInputPopover } from '@elastic/eui';
 
 import { useRangeSlider } from '../embeddable/range_slider_embeddable';
 import { RangeSliderPopover, EuiDualRangeRef } from './range_slider_popover';
 
-import './range_slider.scss';
 import { ControlError } from '../../control_group/component/control_error_component';
-
-const INVALID_CLASS = 'rangeSliderAnchor__fieldNumber--invalid';
+import { RangeValue } from '../../../common/range_slider/types';
+import { RangeSliderButton } from './range_slider_button';
+import './range_slider.scss';
 
 export const RangeSliderControl: FC = () => {
   const rangeRef = useRef<EuiDualRangeRef>(null);
   const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
 
   const rangeSlider = useRangeSlider();
 
-  const min = rangeSlider.select((state) => state.componentState.min);
-  const max = rangeSlider.select((state) => state.componentState.max);
   const error = rangeSlider.select((state) => state.componentState.error);
-  const isInvalid = rangeSlider.select((state) => state.componentState.isInvalid);
-
-  const id = rangeSlider.select((state) => state.explicitInput.id);
-  const value = rangeSlider.select((state) => state.explicitInput.value) ?? ['', ''];
-  const isLoading = rangeSlider.select((state) => state.output.loading);
+  const value = rangeSlider.select((state) => state.explicitInput.value);
+  const [displayedValue, setDisplayedValue] = useState<RangeValue>(value ?? ['', '']);
 
-  const hasAvailableRange = min !== '' && max !== '';
-
-  const hasLowerBoundSelection = value[0] !== '';
-  const hasUpperBoundSelection = value[1] !== '';
+  const debouncedOnChange = useMemo(
+    () =>
+      debounce((newRange: RangeValue) => {
```

Comment:
Nice usage of the lodash debounce to keep some of this in the react component rather than putting it into the embeddable class. Also great to see the debounce added!